lib/core/ogs-tlv-msg.c

```diff
@@ -509,6 +506,7 @@ static int tlv_parse_compound(void *msg, ogs_tlv_desc_t *parent_desc,
     uint32_t offset = 0;
     uint8_t index = 0;
     int i = 0, j;
+    uint8_t count[(mode == OGS_TLV_MODE_T2_L2) ? 65536 : 256][OGS_TLV_MAX_INSTANCE];
```

Comment:
I think this is quite an overkill for the stack because tlv_parse_compound()is recursive, specially for T2 tlvs. It probably makes sense to allocate it in the heap. Opinions?

Reply:
@pespin 

I hope you don't use stack that way. Basically, it would be nice if it could be implemented without such a data structure. If it is unavoidable, of course, it seems like it should be implemented as a heap.

Thanks a lot!
Sukchan

Reply:
I couldn't find another way so far, so I'll probably go for keeping it this way but using the heap.

I believe this patch also fixes some already present bugs in main, for instance in the use of several ogs_pfcp_tlv_desc_fq_csid structures in ogs_pfcp_msg_desc_pfcp_session_establishment_request. Probably without this patch all parsed csids end up in struct ogs_pfcp_session_establishment_request_s field sgw_c_fq_csid.

Maybe you have also a better idea on a better value for OGS_TLV_MAX_INSTANCE, which I took so far without looking in deep.